src/pages/index.js

```diff
@@ -54,69 +60,229 @@ const cardSection = new Section(
+  .then(({ userInfo, cards }) => {
+    userInfor.setUserInfo({
+      name: userInfo.name,
+      description: userInfo.about,
+    });
+    userInfor.updateAvatarImage({ avatar: userInfo.avatar });
+    cardSection.renderItems(cards);
+  })
+  .catch((err) => {
+    console.error("Failed to load user information or cards:", err);
+  });
+
+// // Open avatar edit popup
+const avatarEditButton = document.querySelector(".profile__edit-avatar-button");
+avatarEditButton.addEventListener("click", () => {
+  avatarPopup.open();
+});
+
+// // Function to handle avatar update
+function handleAvatarSubmit(inputData, saveButton) {
+  renderSaving(true, saveButton);
+  api
+    .updateUserAvatar({ avatar: inputData.avatar })
+    .then((userData) => {
+      userInfor.updateAvatarImage({ avatar: userData.avatar });
+      avatarPopup.close();
+    })
+    .catch((err) => {
+      console.error("Error updating avatar:", err);
+    })
+    .finally(() => {
+      renderSaving(false, saveButton);
+    });
+}
+
+function renderSaving(isLoading, buttonElement, defaultText = "Save") {
+  if (isLoading) {
+    buttonElement.textContent = "Saving...";
+  } else {
+    buttonElement.textContent = defaultText;
+  }
+}
+
+//////////////////////////////////// end of project 9 ////////////////////////////////////////////
+
+// //----------------------------------------------------------------------------------------
 //                                     Functions
 //----------------------------------------------------------------------------------------
 function createCard(item) {
-  const card = new Card(item, cardSelector, handleImagePreview);
+  const card = new Card(
+    item,
+    cardSelector,
+    handleImagePreview,
+    (cardId, card) => {
+      deleteCardModal(cardId, card);
+    },
+    (cardId, card) => {
+      handleCardLike(cardId, card);
+    }
+  );
   return card.getview();
 }
 
+// function createCard(item) {
+//   const card = new Card(
+//     item,
+//     cardSelector,
+//     handleImagePreview,
+//     handleCardLike,
+//     (cardId, card) => {
+//       deleteCardModal(cardId, card);
+//     },
+//     (cardId, isLiked, cardElement) => {
+//       handleLikeIcon(cardId, isLiked, cardElement, card);
+//     }
+//   );
+//   return card.getview();
+// }
+
+function handleCardLike(card) {
+  if (!card._isLiked) {
+    api
+      .likeCard(card._id)
+      .then((data) => {
+        console.log(data);
+        card.updateLike(true);
+      })
+      .catch((err) => console.error("Error adding like to card:", err));
+  } else {
+    api
+      .dislikeCard(card._id)
+      .then((data) => {
+        card.updateLike(false);
+      })
+      .catch((err) => console.error("Error removing like from card:", err));
+  }
+}
+
 const editFormValidator = new FormValidator(settings, profileEditForm);
 const addFormValidator = new FormValidator(settings, addCardFormElement);
+const avatarFormValidator = new FormValidator(settings, avatarFormElement);
```

Comment:
**COULD BE IMPROVED**

You can **universally** create instances of validators for all forms in the project storing them inside one object: `formValidators`.  Then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
// define an object for storing validators
const formValidators = {}

const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // Here you get the name of the form (if you don’t have it then you need to add it into each form in `index.html` first)
    const formName = formElement.getAttribute('name')

   // Here you store the validator using the `name` of the form
    formValidators[formName] = validator;
    validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```